sanic/protocols/http2.py

```diff
@@ -0,0 +1,134 @@
+import asyncio
+import io
+import collections
+from functools import partial
+from typing import List, Tuple
+
+from h2.connection import H2Connection
+from h2.events import (
+    ConnectionTerminated, DataReceived, RequestReceived, StreamEnded
+)
+from h2.errors import ErrorCodes
+from h2.exceptions import ProtocolError
+
+from sanic.request import Request
+from sanic.log import log
+
+RequestData = collections.namedtuple('RequestData', ['headers', 'data'])
+
+
+def generate_http2_protocol(ssl, parent_class):
+    ssl.set_ciphers("ECDHE+AESGCM")
```

Comment:
It may be worth adding ChaCha20 here as well: it's safe to use with HTTP/2 and is deployable with OpenSSL 1.1 and higher as well as with LibreSSL.